benchmarks/device_uvector/device_uvector_bench.cu

```diff
@@ -66,4 +74,110 @@ BENCHMARK(BM_ThrustVectorSizeConstruction)
+                  std::is_same_v<Vector, rmm_uvector>,
+                "unsupported vector type");
+  if constexpr (std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector>) {
+    return Vector(num_elements, 0);
+  } else if constexpr (std::is_same_v<Vector, rmm_uvector>) {
+    auto vec = Vector(num_elements, stream);
+    if (zero_init) {
+      cudaMemsetAsync(vec.data(), 0, num_elements * sizeof(int32_t), stream.value());
+    }
+    return vec;
+  }
+}
+
+template <typename Vector>
+int32_t* vector_data(Vector& vec)
+{
+  static_assert(std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector> or
+                  std::is_same_v<Vector, rmm_uvector>,
+                "unsupported vector type");
+  if constexpr (std::is_same_v<Vector, thrust_vector> or std::is_same_v<Vector, rmm_vector>) {
+    return vec.data().get();
+  } else if constexpr (std::is_same_v<Vector, rmm_uvector>) {
+    return vec.data();
+  }
+}
+
+template <typename Vector>
+void vector_workflow(std::size_t num_elements,
+                     int num_blocks,
+                     int block_size,
+                     rmm::cuda_stream const& input_stream,
+                     std::vector<rmm::cuda_stream> const& streams)
+{
+  auto input = make_vector<Vector>(num_elements, input_stream, true);
+  input_stream.synchronize();
+  for (rmm::cuda_stream_view stream : streams) {
+    auto vec = make_vector<Vector>(num_elements, stream);
+    kernel<<<num_blocks, block_size, 0, stream.value()>>>(
+      vector_data(input), vector_data(vec), num_elements);
+  }
+
+  for (rmm::cuda_stream_view stream : streams) {
+    stream.synchronize();
+  }
+}
+
+template <typename Vector>
+void BM_VectorWorkflow(benchmark::State& state)
+{
+  rmm::mr::cuda_async_memory_resource cuda_async_mr{};
+  rmm::mr::set_current_device_resource(&cuda_async_mr);
+
+  rmm::cuda_stream input_stream;
+  std::vector<rmm::cuda_stream> streams(4);
+
+  auto num_elements = state.range(0);
+  int block_size    = 256;
+  int num_blocks    = 16;
+  auto counter      = thrust::make_counting_iterator(0);
+
+  for (auto _ : state) {  // NOLINT(clang-analyzer-deadcode.DeadStores)
+    cuda_event_timer timer(state, true, input_stream);  // flush_l2_cache = true
+    vector_workflow<Vector>(num_elements, num_blocks, block_size, input_stream, streams);
+  }
+
+  constexpr auto num_accesses = 9;
+  auto bytes                  = num_elements * sizeof(int32_t) * num_accesses;
+  state.SetBytesProcessed(static_cast<int64_t>(state.iterations() * bytes));
+
+  rmm::mr::set_current_device_resource(nullptr);
+}
+
+BENCHMARK_TEMPLATE(BM_VectorWorkflow, thrust::device_vector<int32_t>)  // NOLINT
+  ->RangeMultiplier(10)                                                // NOLINT
+  ->Range(100'000, 100'000'000)                                        // NOLINT
+  ->Unit(benchmark::kMicrosecond)
+  ->UseManualTime();
+
+BENCHMARK_TEMPLATE(BM_VectorWorkflow, rmm::device_vector<int32_t>)  // NOLINT
```

Comment:
It may be worth pointing out here that the only difference between `thrust/rmm::device_vector` is that `thrust::device_vector` will be using `cudaMalloc/cudaFree` and not the current memory resource. 

This could have been achieved by using `rmm::device_vector` for both and using `cuda_memory_resource` for one and `cuda_async_memory_resource` for the other.

Reply:
Comment added.